float32 distanceToFace
float32 angleToFace